c7n/resources/redshift.py

```diff
@@ -799,3 +830,60 @@ def process_resource_set(self, resources, tag_keys):
             arn = self.manager.generate_arn(
                 r['ClusterIdentifier'] + '/' + r['SnapshotIdentifier'])
             client.delete_tags(ResourceName=arn, TagKeys=tag_keys)
+
+
+@RedshiftSnapshot.action_registry.register('revoke-access')
+class RedshiftSnapshotRevokeAccess(BaseAction):
+    """Revokes ability of accounts to restore a snapshot
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: redshift-snapshot-revoke-access
+                resource: redshift-snapshot
+                filters:
+                  - type: cross-account
+                    whitelist:
+                      - 012345678910
+                actions:
+                  - type: revoke-access
+    """
+    permissions = ('redshift:RevokeSnapshotAccess',)
+    schema = type_schema('revoke-access')
+
+    def validate(self):
+        for f in self.manager.ctx.policy.data.get('filters', []):
+            if f.get('type') == 'cross-account':
```

Comment:
a better mechanism to check is for f in self.manager.filters: if isinstance(f, RedshiftSnapshotCrossAccount), checking the raw structure means also supporting the use of a 
```
  filters:
    - cross-account
```

in addition to the longer form with parametrization.